components/script/dom/document.rs

```diff
@@ -1368,6 +1369,13 @@ impl Document {
         loader.load_async(load, listener, self);
     }
 
+    pub fn fetch_async(&self, load: LoadType,
+                       request: RequestInit,
+                       fetch_target: IpcSender<FetchResponseMsg>) {
+        let mut loader = self.loader.borrow_mut();
+        loader.fetch_async(load, request, fetch_target, self);
+    }
```

Comment:
I bet the sync XHR code only processes old-style network events, not the new fetch ones.